keps/sig-storage/1432-volume-health-monitor/README.md

```diff
@@ -674,8 +679,180 @@ _This section must be completed when targeting alpha to a release._
   disable this feature is to install or unistall the sidecars, we cannot write
   tests for feature enablement/disablement.
 
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g., what if some components will restart
+   mid-rollout?
+   This feature does not have a feature gate. It is enabled when the health
+   monitoring controller and agent sidecars are deployed with the CSI driver.
+   So the only way for a rollout to fail is that deploying the health
+   monitoring controller or agent sidecars with the CSI driver fails. If
+   the health monitoring controller cannot be deployed, no events on volume
+   condition will be reported on PVCs. If the health monitoring agent cannot
+   be deployed, no event on volume condition will be reported on the pod.
+
+* **What specific metrics should inform a rollback?**
```

Comment:
I think this is important.  How will I know if I an updated health monitoring controller or agent sidecar is malfunctioning?  do they crash/produce events/show metrics that fail?

Reply:
Currently we only report an event on pvc/pod when we have successfully retrieved an abnormal volume condition from the storage system, but when other errors occur in the controller we only log an error but we don't report an event.  We probably should also report an event when other errors occur.  I'll update the KEP with this.